extension/src/experiments/data/index.ts

```diff
@@ -47,43 +52,58 @@ export class ExperimentsData extends BaseData<ExpShowOutput> {
     )
 
     void this.updateAvailableBranchesToSelect(allBranches)
-    const data: ExpShowOutput = []
 
     const { branches, currentBranch } = await this.getBranchesToShowWithCurrent(
       allBranches
     )
+    let gitLog = ''
+    const order: { branch: string; sha: string }[] = []
+    const args: Args = []
 
-    await Promise.all(
-      branches.map(async branch => {
-        const branchFlags = [
-          ExperimentFlag.REV,
-          branch,
-          ExperimentFlag.NUM_COMMIT,
-          this.experiments.getNbOfCommitsToShow(branch).toString()
-        ]
-
-        const output = (await this.expShow(
-          branchFlags,
-          branch
-        )) as ExpShowOutput
-
-        if (branch !== currentBranch) {
-          const workspaceIndex = output.findIndex(
-            exp => exp.rev === EXPERIMENT_WORKSPACE_ID
-          )
-          output.splice(workspaceIndex, 1)
-        }
-        data.push(...output)
-      })
+    for (const branch of branches) {
+      gitLog = await this.collectGitLogAndOrder(gitLog, branch, order, args)
+    }
+
+    const expShow = await this.internalCommands.executeCommand<ExpShowOutput>(
+      AvailableCommands.EXP_SHOW,
+      this.dvcRoot,
+      ...args
     )
 
-    this.collectFiles(data)
+    this.collectFiles({ expShow })
 
-    return this.notifyChanged(data)
+    return this.notifyChanged({ currentBranch, expShow, gitLog, order })
```

Comment:
[F] By doing the least amount of processing possible here we can update the test fixture(s) with raw data and make sure that everything still works.